presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfSequenceEncoding.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata;
+
+import com.facebook.presto.orc.proto.DwrfProto;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class DwrfSequenceEncoding
```

Comment:
In DWRF, sequenceId was intended to be a general concept (although granted I don't know what other use cases there would be for it).  I can make it explicitly for flat maps and it can be updated to be generic if needed in the future.